src/backend/expungeservice/crawler/parsers/case_parser.py

```diff
@@ -39,15 +41,37 @@ def feed(data) -> CaseParserData:
             probation_revoked = date.fromdatetime(datetime.strptime(probation_revoked_date_string, "%m/%d/%Y"))
         else:
             probation_revoked = None  # type: ignore
-        return CaseParserData(district_attorney_number, hashed_charge_data, hashed_dispo_data, balance_due, probation_revoked)
+        return CaseParserData(
+            district_attorney_number, sid, hashed_charge_data, hashed_dispo_data, balance_due, probation_revoked
+        )
 
     @staticmethod
     def __parse_district_attorney_number(soup) -> str:
         DISTRICT_ATTORNEY_KEY = "District Attorney Number:"
         labels = soup.find_all("th", "ssTableHeaderLabel", limit=10)
         table = {tag.string: tag.parent.find("td").string for tag in labels}
         return table.get(DISTRICT_ATTORNEY_KEY, "")
-    
+
+    @staticmethod
+    def __parse_sid(soup) -> str:
+        sid = CaseParser.__attempt_parse_sid(soup)
+        if sid and isinstance(sid, str) and re.match("^[\w-]+$", sid):
+            return sid
+        else:
+            return ""
+
+    @staticmethod
+    def __attempt_parse_sid(soup) -> str:
+        try:
+            party_information = soup.find("div", class_=SECTION_TITLE_CLASS, string="Party Information")
+            for information_section in party_information.parent.next_siblings:
+                sid_text = information_section.find(text=re.compile("SID"))
+                if sid_text:
+                    return sid_text.parent.next_sibling
```

Comment:
I would love to find a better way to reliably parse out the SID. It seems to work for all the test cases. We'll see how it works in the wild.